File: 1402. Reducing Dishes.py
class Solution:
    def maxSatisfaction(self, satisfaction: List[int]) -> int:
        satisfaction.sort(reverse=True)
        summ=0
        res=0
        for i in range(len(satisfaction)):
            summ+=satisfaction[i]
            if summ <0:
                break
            res+=summ
        return res

File: 1402. Reducing Dishes.cpp
class Solution {
public:
    int maxSatisfaction(vector<int>& satisfaction) {
        sort(satisfaction.begin(), satisfaction.end(), greater<int>());
        int sum=0,res=0;
        for (int i=0;i<satisfaction.size();++i){
            sum+=satisfaction[i];
            if(sum<0){
                break;
            }
            res+=sum;
        }
        return res;
    }
};
